notebooks/demo/nx_cugraph_demo.ipynb

```diff
@@ -0,0 +1,672 @@
+    "\n",
+    "To use it, a user sets `NETWORKX_AUTOMATIC_BACKENDS` in their shell to the backend they'd like to use.  If a user has more than one backend installed, the environment variable can also accept a comma-separated list of backends, ordered by priority in which `networkx` should use them, where the first backend that supports a particular API call will be used.  For example:\n",
+    "```\n",
+    "bash> export NETWORKX_AUTOMATIC_BACKENDS=cugraph\n",
+    "bash> python my_nx_app.py  # uses nx-cugraph wherever possible, then falls back to default implementation where it's not.\n",
+    "```\n",
+    "or in the case of multiple backends installed\n",
+    "```\n",
+    "bash> export NETWORKX_AUTOMATIC_BACKENDS=cugraph,graphblas\n",
+    "bash> python my_nx_app.py  # uses nx-cugraph if possible, then nx-graphblas if possible, then default implementation.\n",
+    "```\n",
+    "\n",
+    "`networkx` looks at the environment variable and the installed backends at import time, and will not re-examine the environment after that.  Because `networkx` was already imported in this notebook, the `reimport_nx()` utility will be called after the `os.environ` disctionary is updated to simulate an environment variable being set in the shell.\n",
+    "\n",
+    "**Please note, this is only needed for demonstration purposes to compare runs both with and without fully-automatic backend use enabled.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ[\"NETWORKX_AUTOMATIC_BACKENDS\"] = \"cugraph\"\n",
+    "nx = reimport_networkx()\n",
+    "# reimporting nx requires reinstantiating Graphs since python consideres\n",
+    "# types from the prior nx import != types from the reimported nx\n",
+    "karate_club_graph = nx.karate_club_graph()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Once the environment is updated, re-running the same `betweenness_centrality` call on the same graph requires no code changes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "43.9 ms ± 222 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit global karate_cg_bc_results\n",
+    "karate_cg_bc_results = nx.betweenness_centrality(karate_club_graph)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "We may see that the same computation actually took *longer* using `nx-cugraph`. This is not too surprising given how small the graph is, since there's a small amount of overhead to copy data to and from the GPU which becomes more obvious on very small graphs.  We'll see with a larger graph how this overhead becomes negligable."
```

Comment:
```suggestion
    "We may see that the same computation actually took *longer* using `nx-cugraph`. This is not too surprising given how small the graph is, since there's a small amount of overhead to copy data to and from the GPU which becomes more obvious on very small graphs.  We'll see with a larger graph how this overhead becomes negligible."
```